src/datasets/filesystems/compression/gzip.py

```diff
@@ -0,0 +1,73 @@
+import os
+from typing import Optional
+
+import fsspec
+from fsspec.archive import AbstractArchiveFileSystem
+from fsspec.utils import DEFAULT_BLOCK_SIZE
+
+
+class GZipFileSystem(AbstractArchiveFileSystem):
+    """Read contents of GZIP archive as a file-system with one file inside."""
+
+    root_marker = ""
+    protocol = "gzip"
+
+    def __init__(
+        self,
+        fo: str = "",
+        mode: str = "rb",
+        target_protocol: Optional[str] = None,
+        target_options: Optional[dict] = None,
+        block_size: int = DEFAULT_BLOCK_SIZE,
+        **kwargs,
+    ):
+        """
+        Parameters
+        ----------
+        fo: str
```

Comment:
just a question (not important to address): why is the docstring for this class different to the format we use elsewhere in `datasets` where the type is written as ``param (`type`)``?

Reply:
Yes, I wonder why using the NumPy style just in this docstring, instead of the usual Google style we use in all the other docstrings. 😅

Reply:
It comes from the fsspec docstrings, let me update that ;)